react_app/src/CreationModal.js

```diff
@@ -0,0 +1,109 @@
+import React, { useState } from "react";
+import UploadIndicator from "./UploadIndicator.js";
+
+function CreationModal({ show, close, children }) {
+  // ------STATE MANAGEMENT-------
+  const [files, setFiles] = useState([]);
+  const [datasetName, setDatasetName] = useState("");
+  const [hasName, setHasName] = useState(false);
+  const [infoAvailable, setInfoAvailable] = useState(false);
+  const [uploading, setUploading] = useState(false);
+  const [uploadComplete, setUploadComplete] = useState(false);
+  const delay = ms => new Promise(res => setTimeout(res, ms));
+
+  const showHideClassName = show
```

Comment:
hmmm... this should be a function, right?

Reply:
Looks like you can actually do (in the below):
return {show && (<div...)}

Reply:
ended up doing @zkoogle's suggestion!